client/sites/tools/sftp-ssh/sftp-form.tsx

```diff
@@ -152,7 +147,7 @@ export const SftpForm = ( {
 
 	const handleResetPassword = () => {
 		setPasswordLoading( true );
-		dispatch( resetSftpPassword( siteId, username ) );
+		dispatch( resetSftpPassword( siteId, username as string ) );
```

Comment:
As a follow-up, we could improve the logic in this file, since it currently doesn't take into account the possibility that `username` is not defined.

I suggest adding some checks and return a fallback UI while they are not defined.

Reply:
It's all under a `username &&` check already, I believe that should be enough:

https://github.com/Automattic/wp-calypso/blob/f74022a7ddc4ef096e0810465605c1db54ad273b/client/sites/tools/sftp-ssh/sftp-form.tsx#L340

Or are you referring to something else?

Reply:
The JSX is, but that doesn't mean guarantee TypeScript that `username` in the highlighted line is defined. It's at best an implicit typecase, at at worst a possible edge-case runtime error.